packages/integration-sdk-runtime/src/execution/__tests__/executeIntegration.test.ts

```diff
@@ -956,13 +967,30 @@ describe('executeIntegrationInstance', () => {
       },
     });
 
-    await expect(
-      executeIntegrationInstance(
-        config.logger,
-        config.instance,
-        config.invocationConfig,
-      ),
-    ).rejects.toThrow(/Duplicate _key detected \(_key=key_a\)/);
+    const response = await executeIntegrationInstance(
+      config.logger,
+      config.instance,
+      config.invocationConfig,
+    );
+    expect(response).toMatchObject({
+      integrationStepResults: [
+        {
+          id: 'a',
+          encounteredTypes: ['duplicate_entity'],
+          status: 'success',
+        },
+        {
+          id: 'b',
+          encounteredTypes: [],
+          status: 'failure',
+        },
+      ],
+      metadata: {
+        partialDatasets: {
+          types: [],
```

Comment:
In this case I think `duplicate_entity` should definitely be added to `partial`, right?

Reply:
I don't think it should be in `partialDatasets.types` because the test steps do not declare the types of entities or relationships they will produce.